### ISSUE TITLE
Missing tests for NotImplementedError in t.protocols.basic

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6322]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6322|
|Type|task|
|Created|2013-02-23 16:57:52Z|
|Branch|https://github.com/twisted/twisted/tree/notimplemented-tests-6322|

Several methods of classes belonging to `twisted.protocols.basic` raise `NotImplementedError` and according to coverage these are not tested.

<details><summary>Searchable metadata</summary>

```
trac-id__6322 6322
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_notimplemented_tests_6322 branches/notimplemented-tests-6322
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1361638672000000 1361638672000000
changetime__1361980046000000 1361980046000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [37214]) Branching to 'notimplemented-tests-6322'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [37215]) add tests, news file. refs [#6322](https://github.com/twisted/twisted/issues/6322)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Added tests and forced a [build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/notimplemented-tests-6322). Please review.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince set owner to @thijstriemstra|
|-|-|

  1. The documentation for `IntNTestCaseMixin.NotImplementedError` has an incorrect reference.
  1. `org` as a local variable name? ('proto' would be more descriptive, and consistent with the other tests)
  1. (optional) "When <method> is not overridden in a subclass, calling it raises `NotImplemented`" maybe
Please commit after addressing the above issues.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

(In [37249]) address review comments. refs [#6322](https://github.com/twisted/twisted/issues/6322)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [tom.prince](#note_4):
>   1. The documentation for `IntNTestCaseMixin.NotImplementedError` has an incorrect reference.
>   1. `org` as a local variable name? ('proto' would be more descriptive, and consistent with the other tests)
>   1. (optional) "When <method> is not overridden in a subclass, calling it raises `NotImplemented`" maybe
> Please commit after addressing the above issues.

Thanks for the review. I'll merge when the new [build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/notimplemented-tests-6322) is green.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [37279]) Merge notimplemented-tests-6322: Add tests for the methods that raise NotImplementedError in twisted.protocols.basic.

Author: thijs
Reviewer: tom.prince
Fixes: [#6322](https://github.com/twisted/twisted/issues/6322)
